src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -46,11 +45,25 @@
 public class HighlightPhase extends AbstractComponent implements FetchSubPhase {
 
     private final Highlighters highlighters;
+    private final ImmutableList<Highlighter> sortedHighlighters;
 
     @Inject
     public HighlightPhase(Settings settings, Highlighters highlighters) {
         super(settings);
         this.highlighters = highlighters;
+        // TODO: would be nice if each highlighter would have a precedence and we could just sort here by this value
+        ImmutableList.Builder sortedListBuilder = ImmutableList.builder();
```

Comment:
This is what I meant, yeah. I'd have made it a ```private static final ImmutableList<String>``` instead of ```Immutable<Highlighter>``` but it doesn't make much difference.